### PR TITLE
Convert active cluster check to script, update release/validate chart templates to use script and add working directory inputs to ensure work takes place in the correct directory

### PR DESCRIPTION
**Pull request by MartyFox**
### Change description ###
Convert active cluster check to script, update release/validate chart templates to use script and add working directory inputs to ensure work takes place in the correct directory

Without Working Directory input the Helm steps will fail as they cannot find the files for the calling repo, the templates checkout multiple repos which causes Azure DevOps to create specific folder names for each repo and work must be targeted to the correct folder.

### CODE REVIEW

File: scripts/auto-detect-cluster.sh [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    return 1
+}
+OPTIND=1
+
+while getopts r:k:e: o; do
+    case "$o" in
+        r)
+            aksResourceGroupName="$OPTARG"
+            ;;
+        k)
```
Comment (timja) [inline scripts/auto-detect-cluster.sh:18]: could we add long arguments please and use those instead so they are self explanatory when reading the code?

Code is read many more times than it is written / edited, making it clear to the reader would be better especially given i don't see how / why `k` is even used here?

Comment (MartyFox) [reply scripts/auto-detect-cluster.sh]: Script updated to use `getopt` which allows for long option inputs, inputs given sensible names to make things easier for end users and steps template updated to use long options so its obvious what they are doing.
Working pipeline shown here: `https://dev.azure.com/hmcts/PlatformOperations/_build/results?buildId=477602&view=results`

**Review by timja — COMMENTED** [review]
Could you test with this repository please?
https://github.com/hmcts/chart-neuvector/blob/master/azure-pipelines.yml

It's a special case which has to run in sandbox instead of preview, so it specifies an explicit cluster to use. 

It's been broken by refactoring here before.

**Comment by MartyFox** [comment]
@timja I've tested the `chart-neuvector` pipeline and it works as expected, it skips the the active cluster check as parameters have been supplied by the calling pipeline. Authentication step confirms the use of `CFT-Sbox` AKS cluster.

https://dev.azure.com/hmcts/PlatformOperations/_build/results?buildId=479000&view=logs&j=92058b04-f16a-5035-546c-cae3ad5e2f5f&t=f13bb37c-47f4-5245-a5e2-b6fac2abf4df


Default non parameters:
image

Supplied AKS parameters:
image

**Review by timja — APPROVED** [review]
nice!

**Pull request merged**